### PR TITLE
TRUNK-2809:Removing a description from an existing concept and trying to...

### PR DESCRIPTION
**Pull request by SandeepRaparthi1**
... save should give a user friendly error message

### CODE REVIEW

**Comment by dkayiwa** [comment]
@SandeepRaparthi1 can you squash these commits into one?
Secondly, did you run mvn clean install? Travis reports some failures.

**Comment by SandeepRaparthi1** [comment]
@dkayiwa Have Ensured that the test cases pass now the problem was the test passed at my end but Travis CI failed Apologies my fault Now Everything is fine please review my PR :)

**Comment by SandeepRaparthi1** [comment]
@dkayiwa Made the changes and squashed the commits :) Please review the PR and let me know if i have to do more changes :)

**Comment by SandeepRaparthi1** [comment]
@dkayiwa when the  concept.getDescriptions is NULL we register a field error against descriptions
and i have checked that there is no null pointer exception when we call size() method on it in the test case i have written :)

File: api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java [hunk 1]
```diff
@@ -452,9 +462,85 @@ public void validate_shouldPassIfDifferentConceptsHaveTheSameShortNames() throws
...
+	/**
+	 * @see ConceptValidator#validate(Object,Errors)
+	 **/
+	@Test
+	@Verifies(value = "should Not return a Null Pointer Exception", method = "validate(Object,Errors)")
+	public void validate_shouldNotReturnaNullPointerException() throws Exception {
+		Concept concept = new Concept();
+		Collection<ConceptDescription> descriptions = concept.getDescriptions();
+		Assert.assertTrue(descriptions.isEmpty());
+		Assert.assertNotNull(descriptions.size());
```
Comment (SandeepRaparthi1) [inline api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java]: @dkayiwa here the line added to ensure that there is no null pointer exception :)

Comment (dkayiwa) [inline api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java]: Do
concept.setDescription(null);
Then pass it through the validator and tell me what happens.

On Thu, Feb 26, 2015 at 8:12 PM, Sandeep Raparthi <notifications@github.com>
wrote:

> In api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java
> <https://github.com/openmrs/openmrs-core/pull/1354#discussion_r25444672>:
>
> > +		}
> > +		Errors errors = new BindException(concept, "concept");
> > +		new ConceptValidator().validate(concept, errors);
> > +		Assert.assertTrue(errors.hasErrors());
> > +	}
> > +	
> > +	/**
> > +	 * @see ConceptValidator#validate(Object,Errors)
> > +	 **/
> > +	@Test
> > +	@Verifies(value = "should Not return a Null Pointer Exception", method = "validate(Object,Errors)")
> > +	public void validate_shouldNotReturnaNullPointerException() throws Exception {
> > +		Concept concept = new Concept();
> > +		Collection<ConceptDescription> descriptions = concept.getDescriptions();
> > +		Assert.assertTrue(descriptions.isEmpty());
> > +		Assert.assertNotNull(descriptions.size());
>
> @dkayiwa <https://github.com/dkayiwa> here the line added to ensure that
> there is no null pointer exception :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/1354/files#r25444672>.
>



-- 
If we keep uppermost in our minds the unkind and unjust acts of others, we
shall find it impossible to love them as Christ has loved us; but if our
thoughts dwell upon the wondrous love and pity of Christ for us, the same
spirit will flow out to others.

Comment (SandeepRaparthi1) [inline api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java]: @dkayiwa I have added that line and checked  Test case passes :)

Comment (SandeepRaparthi1) [inline api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java]: @dkayiwa shall i make that change and update the PR ? :)

Comment (dkayiwa) [inline api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java]: Yes i want to see that test as part of your pull request.

On Thu, Feb 26, 2015 at 8:34 PM, Sandeep Raparthi <notifications@github.com>
wrote:

> In api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java
> <https://github.com/openmrs/openmrs-core/pull/1354#discussion_r25446489>:
>
> > +		}
> > +		Errors errors = new BindException(concept, "concept");
> > +		new ConceptValidator().validate(concept, errors);
> > +		Assert.assertTrue(errors.hasErrors());
> > +	}
> > +	
> > +	/**
> > +	 * @see ConceptValidator#validate(Object,Errors)
> > +	 **/
> > +	@Test
> > +	@Verifies(value = "should Not return a Null Pointer Exception", method = "validate(Object,Errors)")
> > +	public void validate_shouldNotReturnaNullPointerException() throws Exception {
> > +		Concept concept = new Concept();
> > +		Collection<ConceptDescription> descriptions = concept.getDescriptions();
> > +		Assert.assertTrue(descriptions.isEmpty());
> > +		Assert.assertNotNull(descriptions.size());
>
> @dkayiwa <https://github.com/dkayiwa> shall i make that change and update
> the PR ? :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/1354/files#r25446489>.
>



-- 
If we keep uppermost in our minds the unkind and unjust acts of others, we
shall find it impossible to love them as Christ has loved us; but if our
thoughts dwell upon the wondrous love and pity of Christ for us, the same
spirit will flow out to others.

Comment (SandeepRaparthi1) [inline api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java]: @dkayiwa Yup Sure 5 minutes :)

**Comment by SandeepRaparthi1** [comment]
@dkayiwa I have made the required changes :) Please review it :)

**Comment by SandeepRaparthi1** [comment]
Closing this because it has merge conflicts and creating a new one!

**Pull request closed without merging**